lib/src/Uncertainty/Algorithm/WeightedExperiments/openturns/SmolyakExperiment.hxx

```diff
@@ -0,0 +1,115 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Abstract top-level view of a Smolyak experiment
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_SMOLYAKEXPERIMENT_HXX
+#define OPENTURNS_SMOLYAKEXPERIMENT_HXX
+
+#include "openturns/WeightedExperimentImplementation.hxx"
+#include "openturns/WeightedExperiment.hxx"
+#include "openturns/Collection.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+
+
+/**
+ * @class SmolyakExperiment
+ *
+ * The class describes the probabilistic concept of Smolyak experiment
+ */
+class OT_API SmolyakExperiment
+  : public WeightedExperimentImplementation
+{
+  CLASSNAME
+public:
+
+  typedef Collection<WeightedExperiment> WeightedExperimentCollection;
+  typedef PersistentCollection<WeightedExperiment> WeightedExperimentPersistentCollection;
+
+  /** Compare two points approximately */
+  static bool comparePointsApproximately(const Point x, const Point y);
+
+  /** Default constructor */
+  SmolyakExperiment();
+
+  /** Parameters constructor */
+  explicit SmolyakExperiment(const WeightedExperimentCollection & experimentCollection, const UnsignedInteger level);
```

Comment:
no need for explicit if level has no default value

Reply:
Surprising, but true :) Fixed.